src/main/kotlin/gay/j10a1n15/customscoreboard/feature/customscoreboard/CustomScoreboardBackground.kt

```diff
@@ -0,0 +1,42 @@
+package gay.j10a1n15.customscoreboard.feature.customscoreboard
+
+import com.mojang.blaze3d.platform.NativeImage
+import gay.j10a1n15.customscoreboard.config.categories.BackgroundConfig
+import net.minecraft.client.renderer.texture.DynamicTexture
+import net.minecraft.resources.ResourceLocation
+import tech.thatgravyboat.skyblockapi.helpers.McClient
+import java.io.File
+
+object CustomScoreboardBackground {
+
+    private val texturepackTexture = ResourceLocation.fromNamespaceAndPath("customscoreboard", "scoreboard.png")
+    private val dynamicTexture = ResourceLocation.fromNamespaceAndPath("customscoreboard", "dynamic/scoreboard")
+
+    private var dynamic = false
+
+    fun load() {
+        runCatching {
+            val file = File(BackgroundConfig.customImageFile)
+            if (file.exists()) {
+                file.inputStream().use { stream ->
+                    val image = NativeImage.read(NativeImage.Format.RGBA, stream)
+                    McClient.tell {
+                        val texture = DynamicTexture({ "Custom Scoreboard Background" }, image)
+                        McClient.self.textureManager.register(dynamicTexture, texture)
+                    }
+                    dynamic = true
+                }
+            } else {
+                dynamic = false
+            }
+        }.onFailure(Throwable::printStackTrace)
+    }
+
+    fun getTexture(): ResourceLocation {
+        return if (dynamic) {
+            dynamicTexture
+        } else {
+            texturepackTexture
+        }
+    }
```

Comment:
```suggestion
        fun getTexture(): ResourceLocation = dynamicTexture.takeIf { dynamic } ?: texturepackTexture
```

or something similar idc